index.bs

```diff
@@ -11302,6 +11306,136 @@ The [=remote end steps=] given |session| and |command parameters| are:
 </div>
 
 
+## The webExtensions Module ## {#module-webExtensions}
+
+The <dfn export for=modules>webExtensions</dfn> module contains functionality for
+managing and interacting with web-extensions.
+
+### Definition ### {#module-webExtensions-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+WebExtensionsCommand = (
+  webExtensions.Install
+)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+WebExtensionsResult = (
+  webExtensions.InstallResult
+)
+</pre>
+
+### Types ### {#module-webExtensions-types}
+
+#### The extensions.Extension Type #### {#type-webExtensions-Extension}
+
+<div algorithm>
+To <dfn>expand a extension data spec</dfn> given |extension data spec|:
+
+1. Let |installation type| be a value of <code>type</code> field of |extension data spec|.
+
+1. If installing |installation type| isn't supported return [=error=] with [=error code=] [=unsupported operation=].
+
+1. Switch on the value of the <code>installation type</code>:
+
+   <dl>
+    <dt>"<code>path</code>"
+    <dd>
+      1. Let |path| be a value of <code>path</code> field of |extension data spec|.
+      1. Let |entry| be a [=directory entry=] represeting the directory with that |path|.
+
+    <dt>"<code>archive-path</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |extension data spec|.
+      1. Perform any implementation-defined steps to extract ZIP from |archive path| and let |entry| by a [=directory entry=] representing the extraction result, if this fails return [=error=] with [=error code=] [=invalid web-extension=].
```

Comment:
May this change be appropriate?

```suggestion
      1. Perform any implementation-defined steps to extract the archive from |archive path| by using the ZIP algorithm, and let |entry| by a [=directory entry=] representing the extraction path, if this fails return [=error=] with [=error code=] [=invalid web-extension=].
```

Do we also have to add that the remote end is responsible for removing the path to not leave the extracted data behind?